go/vt/vttablet/tabletserver/tx_pool.go

```diff
@@ -76,31 +74,39 @@ type (
 // NewTxPool creates a new TxPool. It's not operational until it's Open'd.
 func NewTxPool(env tabletenv.Env, limiter txlimiter.TxLimiter) *TxPool {
 	config := env.Config()
-	transactionTimeout := config.Oltp.TxTimeoutSeconds.Get()
 	axp := &TxPool{
-		env:                env,
-		scp:                NewStatefulConnPool(env),
-		transactionTimeout: sync2.NewAtomicDuration(transactionTimeout),
-		ticks:              timer.NewTimer(transactionTimeout / 10),
-		limiter:            limiter,
-		txStats:            env.Exporter().NewTimings("Transactions", "Transaction stats", "operation"),
+		env:     env,
+		scp:     NewStatefulConnPool(env),
+		ticks:   timer.NewTimer(txKillerTimeoutInterval(config)),
+		limiter: limiter,
+		txStats: env.Exporter().NewTimings("Transactions", "Transaction stats", "operation"),
 	}
 	// Careful: conns also exports name+"xxx" vars,
 	// but we know it doesn't export Timeout.
-	env.Exporter().NewGaugeDurationFunc("TransactionTimeout", "Transaction timeout", axp.transactionTimeout.Get)
+	env.Exporter().NewGaugeDurationFunc("OlapTransactionTimeout", "OLAP transaction timeout", func() time.Duration {
+		return config.TxTimeoutForWorkload(querypb.ExecuteOptions_OLAP)
+	})
+	env.Exporter().NewGaugeDurationFunc("TransactionTimeout", "Transaction timeout", func() time.Duration {
+		return config.TxTimeoutForWorkload(querypb.ExecuteOptions_OLTP)
+	})
 	return axp
 }
 
 // Open makes the TxPool operational. This also starts the transaction killer
 // that will kill long-running transactions.
 func (tp *TxPool) Open(appParams, dbaParams, appDebugParams dbconfigs.Connector) {
 	tp.scp.Open(appParams, dbaParams, appDebugParams)
-	tp.ticks.Start(func() { tp.transactionKiller() })
+	tp.ticks.SetInterval(txKillerTimeoutInterval(tp.env.Config()))
```

Comment:
this is initialized once in Open of txpool, we can remove this setting here.

Reply:
Are you saying "we call `tp.ticks.SetInterval` once in `NewTxPool`", so we can remove it in `Open`?

The reason I have it in both places is because we have some unit tests that first call `NewTxPool`, and then change the TX timeout, and then do `Open`.

I will see if I can simplify things either in tests or in `NewTxPool`, so we only set the interval once.